extensions/jira-search/src/issue.ts

```diff
@@ -79,13 +81,29 @@ function buildJql(query: string): string {
   return jql + " order by lastViewed desc"
 }
 
-function jqlFor(query: string): string {
-  return isIssueKey(query) ? `key=${query}` : buildJql(query)
+function jqlForFilter(filter: IssueFilter) {
+  if (filter === "issuesInOpenSprints") return "sprint in openSprints()"
+  if (filter === "myIssues") return "assignee=currentUser()"
+  if (filter === "myIssuesInOpenSprints") return "assignee=currentUser() AND sprint in openSprints()"
+  return ""
```

Comment:
```suggestion
  switch (filter) {
    case "allIssues": return ""
    case "issuesInOpenSprints": return "sprint in openSprints()"
    case "myIssues": return "assignee=currentUser()"
    case "myIssuesInOpenSprints": return "assignee=currentUser() AND sprint in openSprints()"
  }
```

In my opinion, this is prettier, and it gives us an exhaustive check of the switch-statement for free: If we add new filters to the type but forget to add them to this switch statement, the typescript compiler will complain about the missing cases. That's why I used `case allIssues` instead of `default`.